app/scripts/metamask-controller.js

```diff
@@ -1786,6 +1787,7 @@ export default class MetamaskController extends EventEmitter {
 
     const wasLocked = !isUnlocked
     if (wasLocked) {
+      this.emit('unlocked')
```

Comment:
Is the extension unlocked in this branch? 🤔 I don't follow why that would be the case

Reply:
I think it is, if you look at the statement starting on L1776 above:
```javascript
if (!addresses.length) {
  return
}
```

Reply:
The keyring can still have addresses _and_ be locked though

Reply:
Or at least I think it can? Could be wrong. I looked at the KeyringController and I couldn't see how the presence of addresses would guarantee it was unlocked.

Reply:
This has been fixed by updating `eth-keyring-controller`, which now emits an `unlock` event every time it becomes unlocked.